pyproject.toml

```diff
@@ -7,19 +7,21 @@ name = "advent-of-code-data"
 version = "2.0.1"
 description = "Get your puzzle data with a single import"
 requires-python = ">=3.9"
-classifiers = [
+classifiers = [  # https://pypi.org/classifiers/
     "Intended Audience :: Developers",
-    "Programming Language :: Python :: 3",
     "License :: OSI Approved :: MIT License",
-    "Topic :: Software Development :: Libraries",
+    "Programming Language :: Python :: 3",
     "Topic :: Games/Entertainment :: Puzzle Games",
+    "Topic :: Software Development :: Libraries",
+    "Typing :: Typed",
 ]
 dependencies = [
     "beautifulsoup4",
     "pebble",
     "urllib3",
-    'tzdata ; platform_system == "Windows"',
+    'tzdata; platform_system == "Windows"',
     "aocd-example-parser >= 2023.2",
+    'typing-extensions; python_version < "3.11"',
```

Comment:
As stated above, I'd just make this unconditional. `typing_extensions` is basically part of the reality of Python libraries now.

Reply:
The reason for this was that [`typing.Self`](https://docs.python.org/3/library/typing.html#typing.Self) was not avail in stdlib until 3.11.

Reply:
I know why `typing_extensions` is used. :-) My point is that the whole dance in `_compat` plus the *condition* on `python_version < "3.11"` here is overkill.

It is far simpler to just use `from typing_extensions import Self` in your code and to install `typing_extensions` unconditionally. `aocd` is far from the only library that would have a dependency on`typing_extensions`, in any given Python version, at the moment.